src/sentry/data_export/tasks.py

```diff
@@ -102,8 +103,24 @@ def assemble_download(
 
             processor = get_processor(data_export, environment_id)
 
-            with tempfile.TemporaryFile() as tf:
-                writer = csv.DictWriter(tf, processor.header_fields, extrasaction="ignore")
+            with tempfile.TemporaryFile(mode="w+b") as tf:
+                # XXX(python3):
+                #
+                # In python2 land we write utf-8 encoded strings as bytes via
+                # the csv writer (see convert_to_utf8). The CSV writer will
+                # ONLY write bytes, even if you give it unicode it will convert
+                # it to bytes.
+                #
+                # In python3 we write unicode strings (which is all the csv
+                # module is able to do, it will NOT write bytes like in py2).
+                # Because of this we use the codec getwriter to transform our
+                # file handle to a stream writer that will encode to utf8.
+                if six.PY2:
+                    tfw = tf
+                else:
+                    tfw = codecs.getwriter("utf-8")(tf)
```

Comment:
This is good, in the last implementation I had where in py3 it would open the file in text mode, the `tell` would be incorrect in that it would report the length of text written, NOT the number of bytes written. So it was very possible we might have run into some subtle bugs there.

This doesn't do that since the tell on the file will still correctly produce the number of bytes the stream writer will have written.

Reply:
![so happy](https://media.giphy.com/media/6tSE1vX3YP1F6/giphy.gif)